client/my-sites/themes/validate-filters.js

```diff
@@ -30,5 +31,21 @@ module.exports = function validateFilter( context, next ) {
 	}
 
 	next();
-};
+}
 
+export function validateVertical( context, next ) {
+	const { vertical } = context.params;
+
+	if ( ! vertical ) {
+		return next();
+	}
+
+	if ( ! includes( getSubjects(), vertical ) ) {
+		if ( context.isServerSide ) {
+			return context.res.redirect( '/themes' );
+		}
+		return page.redirect( '/themes' );
```

Comment:
I think the right thing to do would probably be a call to `next( 'route' )`, which skips the rest of the current route def and tries the next matching one. That's an Express feature tho, I'm not sure we're currently mirroring that for `router`. We might have to add something to `server/isomorphic-routing`, akin to the error handling mw stuff there.

```js
diff --git a/client/my-sites/themes/validate-filters.js b/client/my-sites/themes/validate-filters.js
index 77f33f4..ec4bc03 100644
--- a/client/my-sites/themes/validate-filters.js
+++ b/client/my-sites/themes/validate-filters.js
@@ -42,9 +42,9 @@ export function validateVertical( context, next ) {
 
        if ( ! includes( getSubjects(), vertical ) ) {
                if ( context.isServerSide ) {
-                       return context.res.redirect( '/themes' );
+                       return next( 'route' );
                }
-               return page.redirect( '/themes' );
+               // Don't redirect on the client side.
        }
 
        next();
```